modules/programs/zsh.nix

```diff
@@ -30,6 +30,27 @@ let
     vicmd = "bindkey -a";
   };
 
+  mkLinkFarmEntry = name: dir:
+    let
+      env = pkgs.buildEnv {
+        name = "zsh-${name}-env";
+        paths = cfg.oh-my-zsh.customPkgs;
+        pathsToLink = "/share/zsh/${dir}";
+      };
+    in
+      { inherit name; path = "${env}/share/zsh/${dir}"; };
+
+  mkNamedLinkFarmEntry = name: mkLinkFarmEntry name name;
+
+  omz-custom =
+    if length cfg.oh-my-zsh.customPkgs != 0 then
+      pkgs.linkFarm "oh-my-zsh-custom" [
+        (mkNamedLinkFarmEntry "themes")
+        (mkLinkFarmEntry "completions" "site-functions")
+        (mkNamedLinkFarmEntry "plugins")
+      ]
+    else cfg.oh-my-zsh.custom;
```

Comment:
I haven't used `buildEnv` before, but it seems like `pathsToLink` takes an array. So I'm guessing the three calls can be merged into one:
```nix
  omz-custom =
    let
      dirs = [ "plugins" "site-functions" "themes" ];
      mkPath = dir: "/share/zsh/${dir}";
      env = pkgs.buildEnv {
        name = "oh-my-zsh-custom-env";
        paths = cfg.oh-my-zsh.customPkgs;
        pathsToLink = map mkPath dirs;
      };
    in
    if length cfg.oh-my-zsh.customPkgs != 0 then
      pkgs.linkFarm "oh-my-zsh-custom" (map (dir: { name = dir; path = mkPath dir; }) dirs)
    else cfg.oh-my-zsh.custom;
```
Not a suggestion because this needs to be tested and most likely fixed.

Reply:
for oh-my-zsh the `site-functions` folder for zsh should be linked to `$ZSH_CUSTOM/completions`, right?

Reply:
Are you sure about that? I can only find mentions of `$ZSH_CACHE_DIR/completions`.
My understanding from a couple searches through OMZ's docs and code is that it just uses the system's completion paths, except for plugins that modify it themselves.

My guess is we need a second link farm just for `site-completions` and add that link farm to `fpath` explicitly.  
Anyways if you test and say something works I trust you. I don't even use OMZ so I'm not a reference for this part.